framework/kmod.nix

```diff
@@ -1,17 +1,42 @@
-{ config, lib, ... }:
-{
-  options.hardware.framework.enableKmod = lib.mkEnableOption (lib.mdDoc
+{ config, lib, pkgs, ... }:
+let
+  kernel_version_compatible = lib.versionAtLeast config.boot.kernelPackages.kernel.version "6.10";
+in {
+  options.hardware.framework.enableKmod = (lib.mkEnableOption
     "Enable the community created Framework kernel module that allows interacting with the embedded controller from sysfs."
   ) // {
-    # Enable by default if on new enough version of NixOS
-    default = (lib.versionAtLeast (lib.versions.majorMinor lib.version) "24.05");
+    # enable by default on NixOS >= 24.05 and kernel >= 6.10
+    default = lib.and
+      (lib.versionAtLeast (lib.versions.majorMinor lib.version) "24.05")
+      kernel_version_compatible;
   };
 
-  config = lib.mkIf config.hardware.framework.enableKmod {
-    boot.extraModulePackages = with config.boot.kernelPackages; [
+
+  config.boot = lib.mkIf config.hardware.framework.enableKmod {
+    extraModulePackages = with config.boot.kernelPackages; [
       framework-laptop-kmod
     ];
+
     # https://github.com/DHowett/framework-laptop-kmod?tab=readme-ov-file#usage
-    boot.kernelModules = [ "cros_ec" "cros_ec_lpcs" ];
+    kernelModules = [ "cros_ec" "cros_ec_lpcs" ];
+
+    # add required patch if enabled on kernel <6.10
+    kernelPatches = lib.mkIf (!kernel_version_compatible) [
```

Comment:
How important is this kernel module? Because many users might not want to re-compile their kernel. Can we maybe default to a later kernel instead and disable this module if the kernel is too old?

Reply:
this line is also gated by `config.hardware.framework.enableKmod` above, which under this change defaults to `false` if the kernel version is too old. so i think it addresses what you want already